client/src/lib.rs

```diff
@@ -394,6 +408,7 @@ pub struct WorkflowTaskCompletion {
 #[derive(Clone)]
 pub struct ServiceCallInterceptor {
     opts: ClientOptions,
+    headers: Arc<Mutex<HashMap<String, String>>>,
```

Comment:
Should probably doc here that they are only ever meant to be _read_ from here, not written.

Additionally, probably makes a lot of sense for this to be a `RwLock`